src/libxtp/moldenreader.cc

```diff
@@ -0,0 +1,184 @@
+}
+
+// The returned string contains the line with the next section header
+// or, if it is the last section, an empty string.
+std::string MoldenReader::readMOs(Orbitals& orbitals,
+                                  std::ifstream& input_file) const {
+
+  // setup space to store everything
+  Index basis_size = orbitals.getBasisSetSize();
+  int number_of_electrons = 0;
+  if (basis_size == 0) {
+    throw std::runtime_error(
+        "Basis size not set, atoms were not parsed first.");
+  }
+  orbitals.MOs().eigenvalues().resize(basis_size);
+  orbitals.MOs().eigenvectors().resize(basis_size, basis_size);
+
+  // actual parsing
+  std::string line;
+  std::string tempStr;
+  double tempDouble;
+  Index tempIndex;
+  std::istringstream iss(" ");
+  for (int i = 0; i < basis_size; i++) {  // loop over mo's
+    std::getline(input_file, line);       // skip symmetry label
+    // energy line
+    std::getline(input_file, line);
+    iss.str(line);
+    iss.clear();
+    iss >> tempStr >> tempDouble;
+    orbitals.MOs().eigenvalues()[i] = tempDouble;
+    // spin channel line
+    std::getline(input_file, line);
+    iss.str(line);
+    iss.clear();
+    iss >> tempStr >> tempStr;
+    if (tempStr == "Beta") {
+      throw std::runtime_error(
+          "Open shell systems are currently not supported");
+    }
+    // occupation line
+    std::getline(input_file, line);
+    iss.str(line);
+    iss.clear();
+    iss >> tempStr >> tempDouble;
+    number_of_electrons += (int)tempDouble;
+
+    // MO coefficients
+    for (int j = 0; j < basis_size; j++) {  // loop over ao's
+      std::getline(input_file, line);
+      iss.str(line);
+      iss.clear();
+      iss >> tempIndex >> tempDouble;
+      orbitals.MOs().eigenvectors()(j, i) = tempDouble;
+    }
+  }
+
+  orbitals.setNumberOfAlphaElectrons(number_of_electrons);
+  orbitals.setNumberOfOccupiedLevels(number_of_electrons / 2);
+
+  // Get the correct ordering of atomic orbitals
+  // Note that the molden specification specifies its own orbital ordering
+  // Hence we can't use the qmpackage's ordering.
+
+  OrbReorder reorder(_reorderList, _multipliers, true);
+  reorder.reorderOrbitals(orbitals.MOs().eigenvectors(), _basis);
+
+  getline(input_file, line);
+  return line;
+}
+
+void MoldenReader::addBasissetInfo(Orbitals& orbitals) {
+  BasisSet bs;
+  bs.Load(_basisset_name);
+
+  _basis.Fill(bs, orbitals.QMAtoms());
+  orbitals.setDFTbasisName(_basisset_name);
+  orbitals.setBasisSetSize(_basis.AOBasisSize());
+  orbitals.setAuxbasisName(_aux_basisset_name);
+}
+
+void MoldenReader::parseMoldenFile(const std::string& filename,
+                                   Orbitals& orbitals) {
+
+  if (_basisset_name == "") {
+    throw std::runtime_error(
+        "Basisset names should be set before reading the molden file.");
+  }
+
+  std::ifstream input_file(filename);
+  // Check if succesfull
+  if (input_file.fail()) {
+    throw std::runtime_error("Could not open molden file.");
+  }
```

Comment:
I am not happy about that, what would it take to read in the basisset from molden?